server/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -585,6 +588,7 @@ public void restoreFileFromSnapshot(
         BlobStoreIndexShardSnapshot.FileInfo fileInfo,
         ActionListener<Void> listener
     ) {
+        assert hasReferences();
```

Comment:
This is new, I guess orthogonal to the PR, but still valid in all cases?

Reply:
Yep